### DISCUSSION TITLE
Getting CONNECT_TIMEOUT when connecting to the database

### DISCUSSION THREAD

**Discussion by bernatfortet** [supabase/supabase#21789] [Questions]
I'm getting a timeout when trying to connect to the database via TablePlus and also via Drizzle in Next.js.
I'm using the connection pooling string with transaction mode: postgres://postgres.[MY_PROJET_ID]:[MY_PASSWORD]@aws-0-us-east-1.pooler.supabase.com:6543/postgres

But I've also searched and tried every other possible way to connect to the database and all are failing.

What information could I provide to debug this?

**Comment by slawton3** [comment]
Were you able to resolve this?

**Comment by gregory-krieger** [comment]
My solution for CONNECT_TIMEOUT using Drizzle + Supabase + Next.js


Instead of this :

```
import "dotenv/config";

import { drizzle } from "drizzle-orm/postgres-js";
import postgres from "postgres";
import * as schema from "./schema";

const connectionString = process.env.DATABASE_URL;

// Disable prefetch as it is not supported for "Transaction" pool mode
export const client = postgres(connectionString!, {
  prepare: false,
});
export const db = drizzle(client, { schema });
```

Do this :
```
import "dotenv/config";
import { PostgresJsDatabase, drizzle } from "drizzle-orm/postgres-js";
import postgres from "postgres";
import * as schema from "./schema";

const connectionString = process.env.DATABASE_URL;

const drizzleClient = drizzle(
  postgres(connectionString!, {
    prepare: false,
  }),
  { schema },
);

declare global {
  var database: PostgresJsDatabase<typeof schema> | undefined;
}

export const db = global.database || drizzleClient;
if (process.env.NODE_ENV !== "production") global.database = db;

```

**Comment by msgerbush** [comment]
FYI, I was running into this same issue and it turned out to be a problem with importing 'postgres' from npm.  I switched the import to:
`https://deno.land/x/postgresjs/mod.js` and the timeout went away.

**Reply by florianck** [reply]
this worked, thank you!!!!

**Comment by krokhale** [comment]
You just need to make sure you are not instantiating the client every time you import it while developing locally in nextjs, something like this works:

```
import dotenv from 'dotenv';

if (process.env.NODE_ENV !== 'production'){
    const envFile = '../../.env.local';
    dotenv.config({ path: envFile });
}


import { drizzle } from 'drizzle-orm/postgres-js'
import postgres from 'postgres'

declare global {
    var _db: ReturnType<typeof drizzle> | undefined;
}

const createDb = (() => {
    return () => {
        if (!global._db) {
            const client = postgres(String(process.env.NEXT_DB_URL), { prepare: false });
            global._db = drizzle(client);
        }
        return global._db;
    };
})();
const db = createDb()
export default db;

```

**Comment by alextbogdanov** [comment]
For me, the problem was that my supabase password contained symbols like / and & so I had to encode it and it worked:
https://www.urlencoder.org/

**Comment by artschur** [comment]
getting this too:(

**Comment by jmanm** [comment]
I was getting this error too.  
For me the problem was my connection string. I was using the `SUPABASE_DB_URL` environment variable. In live environments, this is set to the [direct connection](https://supabase.com/docs/guides/database/connecting-to-postgres#direct-connection) URL.  
I switched it to the [transaction pooler](https://supabase.com/docs/guides/database/connecting-to-postgres#supavisor-transaction-mode) connection string, available from the "Connect" button, which fixed the issue.

**Reply by franlocus** [reply]
> I was getting this error too. For me the problem was my connection string. I was using the `SUPABASE_DB_URL` environment variable. In live environments, this is set to the [direct connection](https://supabase.com/docs/guides/database/connecting-to-postgres#direct-connection) URL. I switched it to the [transaction pooler](https://supabase.com/docs/guides/database/connecting-to-postgres#supavisor-transaction-mode) connection string, available from the "Connect" button, which fixed the issue.

Same fix, changed from direct to transaction pooler with --> `pooler.supabase.com:6543/postgres?pgbouncer=true`
Using drizzle + supabase + nextjs

**Comment by Mansafreo** [comment]
My connection was just working fine but now I keep getting the connection timeout. I've checked my net and it was okay. I've also opened the Database on the Supabase Dashboard and it loaded without an issue. Now I get the connection timeout on DBeaver as well as my node project. I'm using the pooling string connection and Prisma for a direct DB connection

**Reply by rolandvar** [reply]
Do you have "Enforce SSL on incoming connections"?
I had the same problem as you. I disabled this setting and it started working again.

**Reply by Mansafreo** [reply]
My enforce SSL is off
